packages/swapper/src/swappers/ChainflipSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,372 @@
+  SwapErrorRight,
+  SwapperDeps,
+  TradeQuote,
+  TradeRate,
+} from '../../../types'
+import {
+  type GetEvmTradeQuoteInput,
+  type ProtocolFee,
+  SwapperName,
+  TradeQuoteError,
+} from '../../../types'
+import { getRate, makeSwapErrorRight } from '../../../utils'
+import {
+  CHAINFLIP_BAAS_COMMISSION,
+  CHAINFLIP_BOOST_SWAP_SOURCE,
+  CHAINFLIP_DCA_BOOST_SWAP_SOURCE,
+  CHAINFLIP_DCA_QUOTE,
+  CHAINFLIP_DCA_SWAP_SOURCE,
+  CHAINFLIP_REGULAR_QUOTE,
+  CHAINFLIP_SWAP_SOURCE,
+  chainIdToChainflipNetwork,
+  usdcAsset,
+} from '../constants'
+import type { ChainflipBaasQuoteQuote, ChainflipBaasQuoteQuoteFee } from '../models'
+import { chainflipService } from '../utils/chainflipService'
+import { getEvmTxFees } from '../utils/getEvmTxFees'
+import { getUtxoTxFees } from '../utils/getUtxoTxFees'
+import { isSupportedAssetId, isSupportedChainId } from '../utils/helpers'
+
+const _getTradeQuote = async (
+  input: CommonTradeQuoteInput,
+  deps: SwapperDeps,
+): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+  const {
+    sellAsset,
+    buyAsset,
+    accountNumber,
+    receiveAddress,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
+    affiliateBps: commissionBps,
+  } = input
+
+  if (!isSupportedChainId(sellAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedAssetId(sellAsset.chainId, sellAsset.assetId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: sellAsset.chainId, assetId: sellAsset.assetId },
+      }),
+    )
+  }
+
+  if (!isSupportedAssetId(buyAsset.chainId, buyAsset.assetId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: buyAsset.chainId, assetId: buyAsset.assetId },
+      }),
+    )
+  }
+
+  const sellChainflipChainKey = `${sellAsset.symbol.toLowerCase()}.${
+    chainIdToChainflipNetwork[sellAsset.chainId]
+  }`
+  const buyChainflipChainKey = `${buyAsset.symbol.toLowerCase()}.${
+    chainIdToChainflipNetwork[buyAsset.chainId]
+  }`
+
+  const brokerUrl = deps.config.REACT_APP_CHAINFLIP_API_URL
+  const apiKey = deps.config.REACT_APP_CHAINFLIP_API_KEY
+
+  // Subtract the BaaS fee to end up at the final displayed commissionBps
+  let serviceCommission = parseInt(commissionBps) - CHAINFLIP_BAAS_COMMISSION
```

Comment:
q: wouldn't it be better to use `bnOrZero` over `parseInt`?

Reply:
Isn't commissionBps always just a small integer number? (0-90?)

Reply:
Think `parseInt` is fine @NeOMakinG as `commissionBps` is not a bignumber but rather a smolnumber